### PR TITLE
itch: init at 25.5.1

### LINKED ISSUE #90233: Package request: itch.io app
**Project description**
The itch.io app is a client for the indie game store itch.io.

**Metadata**

* homepage URL: https://itch.io/app
* source URL: https://github.com/itchio/itch
* license: mit
* platforms: linux, darwin

### PR DESCRIPTION
**Pull request by pasqui23**
###### Motivation for this change
Fixes #90233

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by pasqui23** [comment]
However while itch gui run normally, trying to play a windows game results in an error even with wine installed.

I have posted a bug report here:

https://dale.itch.ovh/iK14K85TvgUh5BKFrGSZLSQx3lsnNGNRP9ItBYQh71awuHG1ebiXJT1LApe9FVQg

File: pkgs/games/itch-setup/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchurl {
+    url = "https://broth.itch.ovh/itch-setup/linux-amd64/${version}/unpacked/default";
+    hash = "sha256-bcJKqhgZK42Irx12BIvbTDMb/DHEOEXljetlDokF7x8=";
+    executable = true;
+  };
+
+in
+(writeShellScriptBin pname ''exec ${steam-run}/bin/steam-run ${src} "$@"'') // {
+
+  passthru = { inherit pname version src; };
```
Comment (SuperSandro2000) [inline pkgs/games/itch-setup/default.nix:16]:
```suggestion
```
That is done by default.

File: pkgs/games/itch/default.nix [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    })
+  ];
+
+  propagatedBuildInputs = [
+    itch-setup
+    libnotify
+    nss
+    gtk3
+  ];
+
```
Comment (SuperSandro2000) [inline pkgs/games/itch/default.nix:44]:
```suggestion
```
This shouldn't be required.

File: pkgs/games/itch-setup/default.nix (hunk 1 above)
Comment (pasqui23) [reply pkgs/games/itch-setup/default.nix:16]: https://github.com/NixOS/nixpkgs/blob/93e7e038c35a14f6afa36e768d9fac2afe572c45/pkgs/build-support/trivial-builders.nix#L116

The attribute is called `name`, not `pname` and neither`src` nor `version` arepassed by default.

Comment (SuperSandro2000) [reply pkgs/games/itch-setup/default.nix:16]: oh, yeah only for mkDerivation, not for writeShellScriptBin.

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 157921` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>itch</li>
    <li>itch-setup</li>
  </ul>
</details>

**Review by davidak — APPROVED** [review]
The code looks good to me.

Builds and works.

I tested it with my account. Starting linux games works well. Tested with one game made in Unity engine and one in Godot. Not tested all settings. Some might not work.

Thanks for your contribution! :sparkles:

**Comment by Shados** [comment]
@pasqui23 I've found itch crashes if I attempt to add another install location (Preferences -> Manage install locations -> Add location); the logs don't show anything revealing but I'm going to _guess_ that electron might be missing some dynamically-loaded dependency that is needed for the file picker.

**Comment by pasqui23** [comment]
I will try to look into it when possible,meanwhile please open a bug report so that also other cann look into it and keep track of it

**Comment by timhae** [comment]
I had similar issues (dll import errors) with another game running with wine (v rising server), removing `.cache .config .local .steam .wine` in the game directory and then starting it again resolved my issues, I will try to narrow it down (probably `.wine` is enough), I did not find a bug report about that hence my comment here, the link to the bug report on the dale.itch.io website seems broken as well. I assume this is a wine issue but did not find much at all..

**Pull request merged**